k8s/pkg/registryserver/registry_cache.go

```diff
@@ -135,24 +135,36 @@ func (rc *registryCacheImpl) GetEndpointsByNsm(nsmName string) []*v1.NetworkServ
 func (rc *registryCacheImpl) CreateOrUpdateNetworkServiceManager(nsm *v1.NetworkServiceManager) (*v1.NetworkServiceManager, error) {
 	existingNsm := rc.networkServiceManagerCache.Get(nsm.GetName())
 
-	if existingNsm != nil {
-		nsm.ObjectMeta = existingNsm.ObjectMeta
-		logrus.Infof("NSM with name %v already exist in cache, updating: %v", nsm.GetName(), nsm)
-		return rc.updateNetworkServiceManager(nsm)
-	}
-
-	logrus.Infof("Creating NSM: %v", nsm)
-	createNsm, err := rc.addNetworkServiceManager(nsm)
-	if err != nil || apierrors.IsAlreadyExists(err) {
-		existingNsm, err = rc.clientset.NetworkservicemeshV1().NetworkServiceManagers(rc.nsmNamespace).Get(nsm.Name, metav1.GetOptions{})
-		if err != nil {
-			return existingNsm, err
+	attempt := 0
+	for {
```

Comment:
I suppose we should still have some timeout, line 1 minute, and don' t try doing this infinite.

Reply:
+1 Infinity is a looong time. Let's try to avoid it :)

Reply:
In practice maximum possible amount of attempts is 2, more attempts is really low probability event. And we don't need timeout, second attempt should be done as soon as possible, while we know the most recent state of resource in registry.

Also it is not really infinite loop, if there are errors that we don't expect to get (i.e. connection refused) we will break

Reply:
Ok, in this case it is fine with me.